pulser-core/pulser/sequence/sequence.py

```diff
@@ -364,6 +365,144 @@ def config_slm_mask(self, qubits: Iterable[QubitId]) -> None:
+                + " with reusable channels is not supported."
+            )
+
+        if (new_device.rydberg_level != self._device.rydberg_level) and strict:
+            raise ValueError(
+                "Device match failed because the"
+                + " devices have different Rydberg levels."
+            )
+
+        # Channel match
+        sample_seq = sample(self)
+        channel_match: dict[str, Any] = {}
+        alert_phase_jump = False
+        strict_error_message = ""
+        ch_type_er_mess = ""
+        for o_d_ch_name, o_d_ch_obj in self.declared_channels.items():
+            for n_d_ch_id, n_d_ch_obj in new_device.channels.items():
+                # Find the corresponding channel on the new device
+                # We verify the channel class then
+                # check whether the addressing Global or local
+                basis_match = o_d_ch_obj.basis == n_d_ch_obj.basis
+                addressing_match = (
+                    o_d_ch_obj.addressing == n_d_ch_obj.addressing
+                )
+                if not (basis_match and addressing_match):
+                    channel_match[o_d_ch_name] = None
+                    ch_type_er_mess = (
+                        f"No match for channel {o_d_ch_name}"
+                        + " with the right basis and addressing."
+                    )
+                    continue
+                if n_d_ch_obj.mod_bandwidth != o_d_ch_obj.mod_bandwidth:
+                    channel_match[o_d_ch_name] = None
+                    strict_error_message = (
+                        f"No channel match for channel {o_d_ch_name}"
+                        + " with the right mod_bandwidth."
+                    )
+                    break
+                if n_d_ch_obj.fixed_retarget_t != o_d_ch_obj.fixed_retarget_t:
+                    channel_match[o_d_ch_name] = None
+                    strict_error_message = (
+                        f"No channel match for channel {o_d_ch_name}"
+                        + " with the right fixed_retarget_t."
+                    )
+                    break
+                if not strict:
+                    channel_match[o_d_ch_name] = n_d_ch_id
+                    break
+                ch_samples = sample_seq.channel_samples[o_d_ch_name]
+                ch_sample_phase = ch_samples.phase
+                # Find if there is phase change between pulses or not
+                phase_is_constant = True
+                if ch_sample_phase.size != 0:
+                    phase_is_constant = bool(
+                        np.all(ch_sample_phase == ch_sample_phase[0])
+                    )
+                # Phase_jump_time and clock_period check
+                phase_jump_time_check = (
+                    o_d_ch_obj.phase_jump_time == n_d_ch_obj.phase_jump_time
+                )
+                clock_period_check = (
+                    o_d_ch_obj.clock_period % n_d_ch_obj.clock_period == 0
+                )
```

Comment:
It seems that this doesn't work after all, let's go back to:
```suggestion
                clock_period_check = (
                    o_d_ch_obj.clock_period == n_d_ch_obj.clock_period
                )
```